src/core/thread/mle.cpp

```diff
@@ -3874,7 +3946,270 @@ void Mle::HandleLinkProbe(const Message &aMessage, const Ip6::MessageInfo &aMess
 }
 #endif // OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
 
+void Mle::HandleLinkRequest(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo, Neighbor *aNeighbor)
+{
+    otError error = OT_ERROR_NONE;
+
+    OT_UNUSED_VARIABLE(aMessage);
+    OT_UNUSED_VARIABLE(aNeighbor);
+
+    Log(kMessageReceive, kTypeLinkRequest, aMessageInfo.GetPeerAddr());
+
+#if OPENTHREAD_FTD
+    if (IsRouterOrLeader())
+    {
+        error = Get<MleRouter>().HandleLinkRequest(aMessage, aMessageInfo, aNeighbor);
+    }
+    else
+#endif
+    {
+#if OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+        error = HandleLinkRequest(aMessage, aMessageInfo);
+#endif
+    }
+
+    LogProcessError(kTypeLinkRequest, error);
+}
+
+void Mle::HandleLinkAccept(const Message &         aMessage,
+                           const Ip6::MessageInfo &aMessageInfo,
+                           uint32_t                aKeySequence,
+                           Neighbor *              aNeighbor)
+{
+    otError  error = OT_ERROR_NONE;
+    uint16_t sourceAddress;
+
+    OT_UNUSED_VARIABLE(aKeySequence);
+    OT_UNUSED_VARIABLE(aNeighbor);
+
+    // Source Address
+    SuccessOrExit(error = Tlv::Find<SourceAddressTlv>(aMessage, sourceAddress));
+
+    Log(kMessageReceive, kTypeLinkAccept, aMessageInfo.GetPeerAddr(), sourceAddress);
+
+#if OPENTHREAD_FTD
+    if (IsActiveRouter(sourceAddress))
+    {
+        error = Get<MleRouter>().HandleLinkAccept(aMessage, aMessageInfo, aKeySequence, aNeighbor, false);
+    }
+    else
+#endif
+    {
+#if OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+        error = HandleLinkAccept(aMessage, aMessageInfo, false);
+#endif
+    }
+
+exit:
+    LogProcessError(kTypeLinkAccept, error);
+}
+
+void Mle::HandleLinkAcceptAndRequest(const Message &         aMessage,
```

Comment:
The method is almost the same with `HandleLinkAccept`. Can we add a parameter to `HandleLinkAccept` and implement this method by re-using `HandleLinkAccept`? And we can consider providing a default value for the new parameter we add in `HandleLinkAccept`.